google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -378,17 +496,33 @@ public ReadOnlyTransaction singleUseReadOnlyTransaction(TimestampBound bound) {
     @Override
     public ReadOnlyTransaction readOnlyTransaction() {
       try {
-        return new AutoClosingReadTransaction(delegate.readOnlyTransaction(), this, false);
+        return new AutoClosingReadTransaction(
+            new Supplier<ReadOnlyTransaction>() {
+              @Override
+              public ReadOnlyTransaction get() {
+                return delegate.readOnlyTransaction();
+              }
+            },
+            this,
+            false);
       } catch (Exception e) {
         close();
         throw e;
       }
     }
 
     @Override
-    public ReadOnlyTransaction readOnlyTransaction(TimestampBound bound) {
+    public ReadOnlyTransaction readOnlyTransaction(final TimestampBound bound) {
       try {
-        return new AutoClosingReadTransaction(delegate.readOnlyTransaction(bound), this, false);
+        return new AutoClosingReadTransaction(
```

Comment:
This snippet is used 4 times.  Please create a method for this snippet.

Reply:
Added method to reduce code duplication.